backend/src/test/java/com/phaete/backend/forage/controller/ForageMapItemControllerIntegrationTest.java

```diff
@@ -0,0 +1,211 @@
+							],
+							"popupText": "Test popup"
+						},
+						"quantity": "ABUNDANT",
+						"quality": "EXCELLENT",
+						"dateFound": "never",
+						"notes": "test"
+					}
+			""";
+
+	private final ForageWikiItem forageWikiItem = new ForageWikiItem(
+			"1",
+			"Apple",
+			ForageCategory.FRUIT,
+			ForageSource.TREE,
+			"Apple Tree",
+			ForageSeason.FALL,
+			List.of("url-to-apple-tree-image")
+	);
+
+	private final CustomMarker customMarker = new CustomMarker(
+			"1",
+			new double[] {51.0, 10.0},
+			"test-icon",
+			new int[] {64, 64},
+			new int[] {32, 64},
+			new int[] {0, 64},
+			"Test popup"
+	);
+
+	private final ForageMapItem forageMapItem = new ForageMapItem(
+			"1",
+			forageWikiItem,
+			customMarker,
+			ForageQuantity.ABUNDANT,
+			ForageQuality.EXCELLENT,
+			"never",
+			"test"
+	);
+
+	@Test
+	void createForageMapItem_expectCreated() throws Exception {
+		mockMvc.perform(MockMvcRequestBuilders.post("/api/forageMapItems")
+				.contentType(MediaType.APPLICATION_JSON)
+				.content(forageMapItemDTOJsonObject)
+		)
+				.andExpect(status().isOk())
+				.andExpect(content().json(forageMapItemDTOJsonObject));
+	}
+
+	@Test
+	void findAllForageMapItems_returnEmptyList_onEmpty_DB() throws Exception {
+		mockMvc.perform(MockMvcRequestBuilders.get("/api/forageMapItems"))
+				.andExpect(status().isOk())
+				.andExpect(content().json("[]"));
+	}
+
+	@Test
+	void findAllForageMapItems_returnsForageMapItems_onSuccess() throws Exception {
+		forageWikiItemRepository.save(forageWikiItem);
+		customMarkerRepository.save(customMarker);
+		forageMapItemRepository.save(forageMapItem);
+
+		String forageMapItemsJsonObject = """
+				[
+					{
+						"id": "1",
+						"forageWikiItem": {
+							"name": "Apple",
+							"category": "FRUIT",
+							"source": "TREE",
+							"description": "Apple Tree",
+							"season": "FALL",
+							"imageURLs": [
+							  "url-to-apple-tree-image"
+							]
+						},
+						"customMarker": {
+							"id": "1",
+							"position": [
+								51.0,
+								10.0
+							],
+							"iconUrl": "test-icon",
+							"iconSize": [
+								64,
+								64
+							],
+							"iconAnchor": [
+								32,
+								64
+							],
+							"popupAnchor": [
+								0,
+								64
+							],
+							"popupText": "Test popup"
+						},
+						"quantity": "ABUNDANT",
+						"quality": "EXCELLENT",
+						"dateFound": "never",
+						"notes": "test"
+					}
+				]
+				""";
+
+		mockMvc.perform(MockMvcRequestBuilders.get("/api/forageMapItems"))
+				.andExpect(status().isOk())
+				.andExpect(content().json(forageMapItemsJsonObject));
+	}
+
+	@Test
+	void findAllForageMapItems_returnsAllForageMapItems_excludeItemsWithCustomMarkerNotInDB() throws Exception {
+		forageWikiItemRepository.save(forageWikiItem);
+		forageMapItemRepository.save(forageMapItem);
+
+		mockMvc.perform(MockMvcRequestBuilders.get("/api/forageMapItems"))
+				.andExpect(status().isOk())
+				.andExpect(content().json("[]"));
+
+	}
```

Comment:
Your integration test should include both valid and invalid items. An empty list as expected could have been result of excluding too many items.

Reply:
Thats what I have the other two tests for the findAll-method for. One of which tests the behaviour that valid mapItems are being shown.